coop_memberspace/controllers/main.py

```diff
@@ -0,0 +1,451 @@
+from __future__ import division
+
+import locale
+import logging
+from datetime import date, datetime, timedelta
+
+import odoo
+import pytz
+from odoo import http, fields, _
+from odoo.http import request
+
+_logger = logging.getLogger(__name__)
+import werkzeug
+
+
+class Website(odoo.addons.website.controllers.main.Website):
+    @http.route("/", type="http", auth="public", website=True)
+    def index(self, **kw):
+        user = request.env.user
+
+        # Get members
+        members = (
+            request.env["res.partner"]
+            .sudo()
+            .search(
+                [
+                    ("is_member", "=", True),
+                    (
+                        "cooperative_state",
+                        "not in",
+                        [
+                            "blocked",
+                            "unpayed",
+                            "not_concerned",
+                            "unsubscribed",
+                        ],
+                    ),
+                ]
+            )
+        )
+
+        # Get member status
+        member_status = user.partner_id.get_warning_member_state()
+
+        # Get next shift
+        shift_registration_env = request.env["shift.registration"]
+        today = date.today()
+        shift_registration = shift_registration_env.sudo().search(
+            [
+                ("shift_id.shift_template_id.is_technical", "=", False),
+                ("partner_id", "=", user.partner_id.id),
+                ("state", "!=", "cancel"),
+                (
+                    "date_begin",
+                    ">=",
+                    datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
+                ),
```

Comment:
```suggestion
                ("date_begin", ">=", datetime.now()),
```




This is no longer necessary in `12.0`.
You can just use datetime objects, without converting to string

I'm sure there are other cases like this one. I won't comment all of them but can you spot them and change them?